x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -138,6 +152,78 @@ protected Clock getClock() {
     public static LicenseService getSharedLicenseService() { return licenseService.get(); }
     public static XPackLicenseState getSharedLicenseState() { return licenseState.get(); }
 
+    /**
+     * Checks if the cluster state allows this node to add x-pack metadata to the cluster state,
```

Comment:
can you document that if the cluster state already contains xpack metadata it is always considered "ready"?

Reply:
I've pushed 5f25ea3c14a